### PR TITLE
Add permission and ability to upload trigger attachment

### LINKED ISSUE #759: Add ability to upload a certificate PDF and offer it to achievers
Add the ability to put an attachment with a trigger.

- [ ] Add a permission for "Trigger Attachments"
- [ ] Add the ability for a staff member to attach a PDF to a trigger
- [ ] Display to the participant with the ability to download the attachment
- [ ] Participant should be able to go back and download the attachment later as well
- [ ] Internationalize so that attachments can be uploaded for each available language
- [ ] Add a boolean for `IsCertificate` so that we can show the certificate image (below) if it is, otherwise we can just do a text link.

### PR DESCRIPTION
**Pull request by rprollingerjr**
- Fixes #759

### CODE REVIEW

**Review by k7hpn — CHANGES_REQUESTED** [review]
- Let's try to stick to only one migration per feature PR for something like this
- If a user does not have the `TriggerAttachments` permission they still see the "Replace the attachment" section with the file selector and the "Clear file selection" button - they shouldn't be shown that if they don't have permission to replace or clear the attachment.

File: src/GRA.Controllers/ViewModel/Profile/AttachmentListViewModel.cs [hunk 1]
```diff
@@ -0,0 +1,15 @@
+﻿using System.Collections.Generic;
+using GRA.Controllers.ViewModel.Shared;
+using GRA.Domain.Model;
+
+namespace GRA.Controllers.ViewModel.Profile
+{
+    public class AttachmentListViewModel
```
[the remark this answers is not in the document]
Comment (rprollingerjr) [reply src/GRA.Controllers/ViewModel/Profile/AttachmentListViewModel.cs:7]: Looks neat, done.

File: src/GRA.Data/Model/Attachment.cs [hunk 2]
```diff
@@ -0,0 +1,14 @@
+﻿using System.ComponentModel.DataAnnotations;
+using GRA.Data.Abstract;
+
+namespace GRA.Data.Model
+{
+    public class Attachment : BaseDbEntity
+    {
+        public string FileName { get; set; }
```
[the remark this answers is not in the document]
Comment (rprollingerjr) [reply src/GRA.Data/Model/Attachment.cs:8]: Got it, done

File: src/GRA.Data/Model/Attachment.cs [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+        [Required]
+        public bool? IsCertificate { get; set; } = true;
+
+        public int SiteId { get; set; }
```
[the remark this answers is not in the document]
Comment (rprollingerjr) [reply src/GRA.Data/Model/Attachment.cs:12]: Yes certainly, done

File: src/GRA.Data/Model/Notification.cs [hunk 4]
```diff
@@ -17,5 +17,6 @@ public class Notification : Abstract.BaseDbEntity
         public bool IsAchiever { get; set; }
         public bool IsJoiner { get; set; }
         public int? AvatarBundleId { get; set; }
+        public string AttachmentFilename { get; set; }
```
[the remark this answers is not in the document]
Comment (rprollingerjr) [reply src/GRA.Data/Model/Notification.cs:20]: 100%

File: src/GRA.Domain.Model/Filters/UserLogFilter.cs [hunk 5]
```diff
@@ -2,8 +2,19 @@
 {
     public class UserLogFilter : BaseFilter
     {
-        public bool? HasBadge { get; set; }
+        public UserLogFilter(int? page) : this(page, 12)
```
[the remark this answers is not in the document]
Comment (rprollingerjr) [reply src/GRA.Domain.Model/Filters/UserLogFilter.cs:5]: Felt dangerous. Done

File: src/GRA.Domain.Service/ActivityService.cs [hunk 6]
```diff
@@ -534,6 +538,12 @@ var alreadyDone
 
             // every trigger awards a badge
             var badge = await AwardBadgeAsync(userIdToLog, trigger.AwardBadgeId);
+            Attachment attachment = null;
```
[the remark this answers is not in the document]
Comment (rprollingerjr) [reply src/GRA.Domain.Service/ActivityService.cs:541]: Ternarys look better OK, done

File: src/GRA.Domain.Service/ActivityService.cs [hunk 7]
```diff
@@ -1189,6 +1201,12 @@ private async Task<Badge> AwardBadgeAsync(int userId, int? badgeId)
 
                 // every trigger awards a badge
                 var badge = await AwardBadgeAsync(userId, trigger.AwardBadgeId);
+                Attachment attachment = null;
```
[the remark this answers is not in the document]
Comment (rprollingerjr) [reply src/GRA.Domain.Service/ActivityService.cs:1204]: Yes, finished

**Pull request merged**